### PR TITLE
Config

### PR DESCRIPTION
**Pull request by fluidvanadium**
ZingoConfigBuilder is now fluent and doctested

### CODE REVIEW

File: zingoconfig/src/lib.rs [hunk 1]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
-    pub fn set_wallet_dir(mut self, dir: PathBuf) -> Self {
-        self.wallet_dir = Some(dir);
+    /// the proxy server we download blockchain information from
+    /// # Examples
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use http::Uri;
+    /// assert_eq!(ZingoConfigBuilder::default().set_lightwalletd_uri(("https://zcash.mysideoftheweb.com:19067").parse::<Uri>().unwrap()).lightwalletd_uri.clone().unwrap(), "https://zcash.mysideoftheweb.com:19067");
+    /// ```
+    pub fn set_lightwalletd_uri(&mut self, lightwalletd_uri: http::Uri) -> &Self {
```
Comment (AloeareV) [inline zingoconfig/src/lib.rs]: In order to chain multiple calls this needs to return a `&mut Self`
```suggestion
    pub fn set_lightwalletd_uri(&mut self, lightwalletd_uri: http::Uri) -> &mut Self {
```

File: zingoconfig/src/lib.rs [hunk 2]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
     }
-    pub fn set_lightwalletd(mut self, lightwalletd_uri: http::Uri) -> Self {
-        self.lightwalletd_uri = Some(lightwalletd_uri);
+    /// # Examples
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use zingoconfig::ChainType::Testnet;
+    /// assert_eq!(ZingoConfigBuilder::default().set_chain(Testnet).create().chain, Testnet);
+    /// ```
+    pub fn set_chain(&mut self, chain: ChainType) -> &Self {
```
Comment (AloeareV) [inline zingoconfig/src/lib.rs]: As above
```suggestion
    pub fn set_chain(&mut self, chain: ChainType) -> &mut Self {
```

File: zingoconfig/src/lib.rs [hunk 3]
```diff
@@ -116,12 +116,38 @@ pub struct ZingoConfig {
...
+    /// the proxy server we download blockchain information from
+    /// # Examples
+    /// ```
+    /// use zingoconfig::ZingoConfigBuilder;
+    /// use tempdir::TempDir;
+    /// let dir = TempDir::new("zingo_doc_test").unwrap().into_path();
+    /// let config = ZingoConfigBuilder::default().set_wallet_dir(dir.clone()).create();
+    /// assert_eq!(config.wallet_dir.clone().unwrap(), dir);
+    /// ```
+    pub fn set_wallet_dir(&mut self, dir: PathBuf) -> &Self {
```
Comment (AloeareV) [inline zingoconfig/src/lib.rs]: ^^
```suggestion
    pub fn set_wallet_dir(&mut self, dir: PathBuf) -> &mut Self {
```

**Review by AloeareV — DISMISSED** [review]

**Review by zancas — DISMISSED** [review]

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]
@AloeareV 's comments have not been resolved

**Review by Oscar-Pepper — DISMISSED** [review]
wonderful

**Review by Oscar-Pepper — DISMISSED** [review]

**Review by Oscar-Pepper — DISMISSED** [review]

**Review by Oscar-Pepper — APPROVED** [review]

**Pull request merged**